tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -3901,9 +3901,13 @@ def __init__(self, input_dataset, batch_size, drop_remainder):
     self._drop_remainder = ops.convert_to_tensor(
         drop_remainder, dtype=dtypes.bool, name="drop_remainder")
 
-    constant_drop_remainder = tensor_util.constant_value(self._drop_remainder)
     # pylint: disable=protected-access
-    if constant_drop_remainder:
+    try:
+      batch_modulo = self._input_dataset.__len__() % self._batch_size
```

Comment:
It's tricky how sometime batch_modulo is a number and sometimes a bool. Can we call it `evenly_divisible` and set it to `self._input_dataset.__len__() % self._batch_size == 0`?